File: skeleton/src/main/java/me/shkschneider/skeleton/helpers/SystemProperties.java
package me.shkschneider.skeleton.helpers;

import android.support.annotation.NonNull;

public class SystemProperties {

    private static String property(@NonNull final String property) {
        final String systemProperty = System.getProperty(property);
        if (systemProperty == null) {
            LogHelper.warning("SystemProperty was NULL");
            return null;
        }

        return systemProperty;
    }

    public static String javaVmName() {
        return property("java.vm.name");
    }
    public static String javaVmVendor() {
        return property("java.vm.vendor");
    }
    public static String javaVmVersion() {
        return property("java.vm.version");
    }
    public static String javaHome() {
        return property("java.home");
    }
    public static String userDir() {
        return property("user.dir");
    }
    public static String userRegion() {
        return property("user.region");
    }
    public static String javaIoTmpdir() {
        return property("java.io.tmpdir");
    }
    public static String javaRuntimeName() {
        return property("java.runtime.name");
    }
    public static String httpAgent() {
        return property("http.agent");
    }
    public static String fileSeparator() {
        return property("file.separator");
    }
    public static String fileEncoding() {
        return property("file.encoding");
    }
    public static String lineSeparator() {
        return property("line.separator");
    }
    public static String osArch() {
        return property("os.arch");
    }
    public static String osName() {
        return property("os.name");
    }
    public static String osVersion() {
        return property("os.version");
    }
    public static String pathSeparator() {
        return property("path.separator");
    }

}

File: kotlin/src/main/kotlin/me/shkschneider/skeleton/kotlin/helper/SystemProperties.kt
package me.shkschneider.skeleton.kotlin.helper

import me.shkschneider.skeleton.kotlin.log.Logger

// <https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html>
object SystemProperties {

    const val FILE_SEPARATOR = "file.separator"
    const val JAVA_CLASS_PATH = "java.class.path"
    const val JAVA_HOME = "java.home"
    const val JAVA_VENDOR = "java.vendor"
    const val JAVA_VENDOR_URL = "java.vendor.url"
    const val JAVA_VERSION = "java.version"
    const val LINE_SEPARATOR = "line.separator"
    const val OS_ARCH = "os.arch"
    const val OS_NAME = "os.name"
    const val OS_VERSION = "os.version"
    const val PATH_SEPARATOR = "path.separator"
    const val USER_DIR = "user.dir"
    const val USER_HOME = "user.home"
    const val USER_NAME = "user.name"

    fun get(property: String): String? =
        System.getProperty(property) ?: run {
            Logger.warning("SystemProperty was NULL")
            return null
        }

}
